src/libutil/filesystem.cpp

```diff
@@ -778,12 +781,19 @@ Filesystem::last_write_time(string_view path, std::time_t time) noexcept
     times.modtime = time;
     _wutime(u8path(path).c_str(), &times);
 #else
+#ifndef __APPLE__
     struct timespec times[2];
     times[0].tv_sec  = 0;
     times[0].tv_nsec = UTIME_OMIT;
     times[1].tv_sec  = time;
     times[1].tv_nsec = 0;
-    utimensat((int)AT_FDCWD, u8path(path).c_str(), times, AT_SYMLINK_NOFOLLOW);
+utimensat((int)AT_FDCWD, u8path(path).c_str(), times, AT_SYMLINK_NOFOLLOW);
+#else
+    struct utimbuf times;
+    times.actime  = time;
+    times.modtime = time;
+    utime(u8path(path).c_str(), &times);
+#endif
```

Comment:
That's a bit of an awkward construction. Let's reverse the sense of this, so it's

```
#ifdef _WIN32
   ... windows case ...
#elif defined(__APPLE__)
   ... apple case ...
#else
    ... default unix case ...
#endif
```

I'm confused about why this is necessary. utimensat is POSIX, should be fine on Apple, and indeed on the Mac laptop I'm using this very moment, "man utimensat" shows me the right thing.

On the other hand, `utime()` is also POSIX, also exists on Linux, and is simpler, so I'm wondering if maybe the simpler solution is not to split apple from linux, and just use your new utime call for both? That is, maybe this is the best construct:

```
void
Filesystem::last_write_time(string_view path, std::time_t time) noexcept
{
#ifdef _WIN32
    struct _utimbuf times;
    times.actime  = time;
    times.modtime = time;
    _wutime(u8path(path).c_str(), &times);
#else
    struct utimbuf times;
    times.actime  = time;
    times.modtime = time;
    utime(u8path(path).c_str(), &times);
#endif
}
```

will that work for both Apple and Linux just fine?